students/WesWang/lesson8/circle.py

```diff
@@ -0,0 +1,51 @@
+import math
+
+class Circle:
+    def __init__(self, radius):
+        self.radius = radius
+    def __str__(self):
+        return "Circle with radius: " + str(self._radius)
+    def __repr__(self):
+        return f"Circle({self._radius})"
+    def __add__(self, other):
+        total_radius = self._radius + other._radius
+        return Circle(total_radius)
+    def __mul__(self, other):
+        if type(other) == Circle:
+            mul_radius = self._radius * other._radius
+        else:
+            mul_radius = self._radius * other
+        return Circle(mul_radius)
+    def __rmul__(self, value):
+        mul_radius = self._radius * value
+        return Circle(mul_radius)
+    def __eq__(self, other):
+        return self._radius == other._radius
+    def __ne__(self, other):
+        return not self._radius == other._radius
+    def __lt__(self, other):
+        return self._radius < other._radius
+    def __gt__(self, other):
+        return not self._radius < other._radius
+    def __le__(self, other):
+        return self._radius >= other._radius
+    def __ge__(self, other):
+        return not self._radius >= other._radius
+    @property
```

Comment:
no reason to have a property that simply gets and sets a attribute. YOu can just use self.radius everywhere, and not bother with the property.

Use a property when there is something to be done in the getter and setter -- like compute the diameter :-)